### PR TITLE
Add CI validation for new master

### PR DESCRIPTION
**Pull request by jojo-stripe**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
     "lint": "next lint && npm run prettier-check",
     "prettier": "prettier './**/*.{js,ts,md,html,css,tsx}' --write",
     "prettier-check": "prettier './**/*.{js,ts,md,html,tsx}' --check",
-    "typescript": "tsc -p ."
+    "typescript": "tsc -p .",
+    "validate-change": "yarn build"
```
Comment (jorgea-stripe) [inline package.json:13]: `validate-change` should ideally be lint + typescript. Then `build` should be run separately after, any reason we aren't doing it that way? `build` is producing the assets, so should not be part of validation!

**Review by jorgea-stripe — COMMENTED** [review]
r? @jojo left some feedback!

File: package.json (hunk 1 above)
Comment (jojo-stripe) [reply package.json:13]: https://github.com/stripe/stripe-connect-furever-demo/blob/4b34816cf42224d3957ed844e650517c76d932e4/package.json#L21 I had just copied it over from here, but will make that change

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -23,6 +12,5 @@ jobs:
       - uses: actions/setup-node@v1
         with:
           node-version: 20
-      - run: npm install --frozen-lockfile
-      - run: npm run build
-      - run: npm run lint
+      - run: yarn install --frozen-lockfile
+      - run: yarn run validate-change
```
Comment (jorgea-stripe) [inline .github/workflows/test.yml:16]:
```suggestion
      - run: yarn run validate-change
      - run: yarn run build
```

Comment (jorgea-stripe) [reply .github/workflows/test.yml:16]: Can we also add `build` separately? Or as part of validate change? I think there are scenarios in which TS build can pass but next build may fail, just want to cover those!

**Review by jorgea-stripe — APPROVED** [review]
Just one minor suggestion!

**Pull request merged**